providers/implementations/signature/rsa_sig.c

```diff
@@ -188,22 +188,36 @@ static void *rsa_newctx(void *provctx, const char *propq)
     prsactx->libctx = PROV_LIBCTX_OF(provctx);
     prsactx->flag_allow_md = 1;
     prsactx->propq = propq_copy;
-    /* Maximum for sign, auto for verify */
-    prsactx->saltlen = RSA_PSS_SALTLEN_AUTO;
+    /* Maximum until digest lnegth for sign, auto for verify */
+    prsactx->saltlen = RSA_PSS_SALTLEN_AUTO_DIGEST_MAX;
```

Comment:
I am worried about this default change especially in the combination with this value being disallowed for verifying with restricted PSS keys?

Reply:
This is not a new issue. The current default seems to be RSA_PSS_SALTLEN_AUTO, and that is also not accepted with restricted PSS keys due to this part:

```c
if (rsa_pss_restricted(prsactx)) {
    switch (saltlen) {
    case RSA_PSS_SALTLEN_AUTO:
        if (prsactx->operation == EVP_PKEY_OP_VERIFY) {
            ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_SALT_LENGTH,
                           "Cannot use autodetected salt length");
            return 0;
        }
        break;
    // …
   }
}
```

Reply:
OK